shell.nix

```diff
@@ -0,0 +1,29 @@
+# A shell to get tooling for Nixpkgs development
+#
+# Note: We intentionally don't use Flakes here,
+# because it would require copying all of Nixpkgs every time the shell is refreshed.
```

Comment:
Can you clarify where this copy would be created? I only see a fetchTarball below, so this also copies nixpkgs bascially.

Reply:
Maybe the comment is referring to how flake repos are copied to the nix store before being evaluated?

Reply:
Yeah, running `nix develop` will copy the entire tree into the store, and when you change _any_ file in Nixpkgs and run `nix develop` again, it creates another copy.

This might work on smaller repos, but with Nixpkgs, these copies are like 500MB each.

I'll improve the wording a bit.

Reply:
Ah, yeah. In that case, we would need to put the flake.nix in a subdirectory.

Reply:
@Mic92 That doesn't work, it still copies the entire Git repo